File: projects/order-creation/core/src/main/java/com/robinfood/core/dtos/transactionrequestdto/DeviceDTO.java
package com.robinfood.core.dtos.transactionrequestdto;

import java.io.Serializable;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

import static com.robinfood.core.constants.GlobalConstants.TIMEZONE_BY_DEVICE_DEFAULT;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class DeviceDTO implements Serializable {

    private static final long serialVersionUID = -2408423642183839382L;

    @NotBlank
    private String ip;

    @NotNull
    private Long platform;

    private String timezone;

    @NotBlank
    private String version;

    public String getTimezone() {

        if (timezone == null || timezone.isEmpty()) {
            return TIMEZONE_BY_DEVICE_DEFAULT;
        }

        return this.timezone;
    }
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/dtos/transactionrequest/DeviceDTO.kt
package com.robinfood.core.dtos.transactionrequest

import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

data class DeviceDTO (

    @NotBlank
    val ip: String,

    @NotNull
    val platform: Int,

    @NotBlank
    val timezone: String,

    @NotBlank
    val version: String
)
